completion/available/composer.completion.bash

```diff
@@ -1,133 +1,136 @@
-#!/usr/bin/env bash
+# shellcheck shell=bash
+cite "about-completion"
+about-completion "composer completion"
 
-_composer()
-{
-    local cur script coms opts com
-    COMPREPLY=()
-    _get_comp_words_by_ref -n : cur words
+function _composer() {
+	local cur script coms opts com
+	COMPREPLY=()
+	_get_comp_words_by_ref -n : cur words
 
-    # for an alias, get the real script behind it
-    if [[ $(type -t ${words[0]}) == "alias" ]]; then
-        script=$(alias ${words[0]} | sed -E "s/alias ${words[0]}='(.*)'/\1/")
-    else
-        script=${words[0]}
-    fi
+	# for an alias, get the real script behind it
+	if [[ $(type -t "${words[0]}") == "alias" ]]; then
+		script=$(alias "${words[0]}" | sed -E "s/alias ${words[0]}='(.*)'/\1/")
+	else
+		script=${words[0]}
+	fi
```

Comment:
I worry that this section might break, but I don't even know what `composure` refers to :joy:  (maybe we should update the `about-completion`).  If it does break, I recommend ripping it out anyways. It's using `sed` to resolve a command from an alias, when one can just use `command`.

Reply:
TIL about the [${BASH_ALIASES}](https://www.gnu.org/software/bash/manual/html_node/Bash-Variables.html) variable:
```
BASH_ALIASES

An associative array variable whose members correspond to the internal
list of aliases as maintained by the alias builtin. (see Bourne Shell Builtins).

Elements added to this array appear in the alias list; however, unsetting array 
elements currently does not cause aliases to be removed from the alias list.

If BASH_ALIASES is unset, it loses its special properties, even if it is
subsequently reset.
```

If I was looking to extract the value of an alias, I would probably use this.

Just the same, it *feels* like the concept of resolving the executable from an alias for completions should be a solved problem as every completion likely has the same concerns.

So I'm wondering how other completions go about this ...

[time passes ...]

OK so I now believe that, in general, completions *don't* worry about trying to deal with being called from aliases.

This is likely why the 'alias-completion` plugin exists.

Which has me thinking: How does the alias completion plugin solve for this?

Well it turns out to be a MUCH more complicated procedure.

So with all that, I'm inclined to feel that this logic should be removed.

**BUT EVEN MORE THAN THAT** : I can't find where this `script` variable is being used after being defined?

So it *looks* like we could probably remove all references to it.

Reply:
@davidpfarrell @cornfeedhobo I have removed this part, thanks for taking your time and reviewing!